app/src/main/java/org/p2p/wallet/newsend/NewSendPresenter.kt

```diff
@@ -327,18 +313,27 @@ class NewSendPresenter(
         val token = token ?: error("Token cannot be null!")
         val address = recipientAddress.addressState.address
         val currentAmount = calculationMode.getCurrentAmount()
+        val currentAmountUsd = calculationMode.getCurrentAmountUsd()
         val lamports = calculationMode.getCurrentAmountLamports()
 
         // the internal id for controlling the transaction state
         val internalTransactionId = UUID.randomUUID().toString()
 
+        val total = feeRelayerManager.buildTotalFee(
+            sourceToken = requireToken(),
+            calculationMode = calculationMode
+        )
+
         appScope.launch {
             try {
-                val destinationAddressShort = address.cutMiddle()
-                val data = ShowProgress(
-                    title = R.string.send_transaction_being_processed,
-                    subTitle = "${currentAmount.toPlainString()} ${token.tokenSymbol} → $destinationAddressShort",
-                    transactionId = emptyString()
+                val data = NewShowProgress(
+                    date = Date(),
+                    tokenUrl = token.iconUrl.orEmpty(),
+                    amountTokens = "${currentAmount.toPlainString()} ${token.tokenSymbol}",
+                    amountUsd = currentAmountUsd.asNegativeUsdTransaction(),
+                    recipient = recipientAddress.nickNameOrAddress(),
+                    totalFee = if (total.sendFee != null) total.getFeesCombined(colorMountain)
+                    else resources.getString(R.string.transaction_transaction_fee_free_value)
```

Comment:
just pass `total` object and you'll be able to delete `colorMountain` and condition check here.
Make check in a new progress bottom sheet